pkg/service/v0/service.go

```diff
@@ -20,39 +26,92 @@ type Service interface {
 	Dummy(http.ResponseWriter, *http.Request)
 }
 
-// Konnectd defines implements the business logic for Service.
-type Konnectd struct {
-	config *config.Config
-	mux    http.Handler
-}
-
 // NewService returns a service implementation for Service.
 func NewService(opts ...Option) Service {
 	ctx := context.Background()
 	options := newOptions(opts...)
 	logger := options.Logger.Logger
-	initKonnectInternalEnvVars(logger)
+	assetVFS := assets.New(
+		assets.Logger(options.Logger),
+		assets.Config(options.Config),
+	)
+
+	if err := initKonnectInternalEnvVars(); err != nil {
+		logger.Fatal().Err(err).Msg("could not initialize env vars")
+	}
+
+	if err := createConfigsIfNotExist(assetVFS); err != nil {
+		logger.Fatal().Err(err).Msg("could not create default config")
+	}
 
 	bs, err := bootstrap.Boot(ctx, &options.Config.Konnectd, &kcconfig.Config{
-		Logger: log.Wrap(logger),
+		Logger: logw.Wrap(logger),
 	})
 
 	if err != nil {
-		logger.Fatal().Err(err).Msg("Could not bootstrap konnectd")
+		logger.Fatal().Err(err).Msg("could not bootstrap konnectd")
 	}
 
 	managers := bs.Managers()
 	routes := []server.WithRoutes{managers.Must("identity").(server.WithRoutes)}
 	handlers := managers.Must("handler").(http.Handler)
 
-	return Konnectd{
+	svc := Konnectd{
+		logger: options.Logger,
 		config: options.Config,
-		mux:    newMux(ctx, routes, handlers, options.Middleware),
 	}
+
+	svc.initMux(ctx, routes, handlers, options)
+
+	return svc
+}
+
+func createConfigsIfNotExist(assets http.FileSystem) error {
+	index, err := os.Create("./index.html")
+	if err != nil {
+		return err
+	}
+
+	index.Close()
+
+	if _, err := os.Stat("./config"); os.IsNotExist(err) {
+		if err := os.Mkdir("./config", 0777); err != nil {
+			return err
+		}
+	}
+
+	if _, err := os.Stat("./config/identifier-registration.yaml"); os.IsNotExist(err) {
+		defaultConf, err := assets.Open("/identifier-registration.yaml")
+		if err != nil {
+			return err
+		}
+
+		defer defaultConf.Close()
+
+		confOnDisk, err := os.Create("./config/identifier-registration.yaml")
+		if err != nil {
+			return err
+		}
+
+		defer confOnDisk.Close()
+
+		conf, err := ioutil.ReadAll(defaultConf)
+		if err != nil {
+			return err
+		}
+
+		err = ioutil.WriteFile("./config/identifier-registration.yaml", conf, 0777)
```

Comment:
@butonic @DeepDiver1975 Hmhhh the perm should also be changed. Any recommendations?

Reply:
It should belong to the owner of the process, and since it contains secrets it should only be readable to that user. Execution is not necessary. But the owner should be able to write the file. Create it with `0600`. If the admin wants to make the file readonly to the process he needs to change the owner and set different permissions anyway.